SmithyDafnyMakefile.mk

```diff
@@ -151,20 +149,16 @@ dafny-reportgenerator:
 clean-dafny-report:
 	rm TestResults/*.csv
 
+check_dafny_version:
```

Comment:
nit, but I'd name this `get_dafny_version` since it's not just a check, it actually sets DAFNY_VERSION which is read by other targets.

And once this is merged we should broadcast that you don't have to set DAFNY_VERSION yourself any more!